proto/concept.proto

```diff
@@ -0,0 +1,963 @@
+        }
+        message Res {
+            optional AttributeType attribute_type = 1;
+        }
+    }
+
+    message PutEntityType {
+        message Req {
+            string label = 1;
+        }
+        message Res {
+            EntityType entity_type = 1;
+        }
+    }
+
+    message PutRelationType {
+        message Req {
+            string label = 1;
+        }
+        message Res {
+            RelationType relation_type = 1;
+        }
+    }
+
+    message PutAttributeType {
+        message Req {
+            string label = 1;
+            AttributeType.ValueType value_type = 2;
+        }
+        message Res {
+            AttributeType attribute_type = 1;
+        }
+    }
+
+    message GetEntity {
+        message Req {
+            bytes iid = 1;
+        }
+        message Res {
+            optional Entity entity = 1;
+        }
+    }
+
+    message GetRelation {
+        message Req {
+            bytes iid = 1;
+        }
+        message Res {
+            optional Relation relation = 1;
+        }
+    }
+
+    message GetAttribute {
+        message Req {
+            bytes iid = 1;
+        }
+        message Res {
+            optional Attribute attribute = 1;
+        }
+    }
+
+    message GetSchemaExceptions {
+        message Req {}
+        message Res {
+            repeated Exception exceptions = 1;
+        }
+    }
+}
+
+message Exception {
+    string code = 1;
+    string message = 2;
+}
+
+message Concept {
+    oneof concept {
+        EntityType entity_type = 1;
+        RelationType relation_type = 2;
+        AttributeType attribute_type = 3;
+
+        RoleType role_type = 4;
+
+        Entity entity = 5;
+        Relation relation = 6;
+        Attribute attribute = 7;
+
+        ThingType.Root root_thing_type = 10;
+    }
+}
+
+message Thing {
+    oneof thing {
+        Entity entity = 1;
+        Relation relation = 2;
+        Attribute attribute = 3;
+    }
+
+    message Req {
+        bytes iid = 1;
+        oneof req {
+            Thing.Delete.Req thing_delete_req = 100;
+            Thing.GetHas.Req thing_get_has_req = 101;
+            Thing.SetHas.Req thing_set_has_req = 102;
+            Thing.UnsetHas.Req thing_unset_has_req = 103;
+            Thing.GetRelations.Req thing_get_relations_req = 104;
+            Thing.GetPlaying.Req thing_get_playing_req = 105;
+
+            Relation.AddPlayer.Req relation_add_player_req = 200;
+            Relation.RemovePlayer.Req relation_remove_player_req = 201;
+            Relation.GetPlayers.Req relation_get_players_req = 202;
+            Relation.GetPlayersByRoleType.Req relation_get_players_by_role_type_req = 203;
+            Relation.GetRelating.Req relation_get_relating_req = 204;
+
+            Attribute.GetOwners.Req attribute_get_owners_req = 300;
+        }
+    }
+
+    message Res {
+        oneof res {
+            Thing.Delete.Res thing_delete_res = 100;
+            Thing.SetHas.Res thing_set_has_res = 102;
```

Comment:
Let's make the numbering blocks sequential, I think the naming lining up is enough.
eg. this section should be
```
... = 100
... = 101
... = 102

... = 200
... = 201
...
```